test/test-TabTracker.js

```diff
@@ -232,6 +232,35 @@ exports.test_TabTracker_refresh = function*(assert) {
   checkLoadUnloadReasons(assert, pingData, loadReasons, unloadReasons);
 };
 
+exports.test_TabTracker_action_pings = function*(assert) {
+  let pingData = {};
+  let userEventPromise = new Promise(resolve => {
+    function observe(subject, topic, data) {
+      if (topic === "user-action-event") {
+        pingData = JSON.parse(data);
+        Services.obs.removeObserver(observe, "user-action-event");
+        resolve();
```

Comment:
nit:  You may want to return the ping in resolve as in:
```
resolve(JSON.parse(data));
```
and later in the code you can say this:
```
let pingData = yield userEventPromise;
```